File: Top Interview Questions/Strings/Longest Common Prefix/Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str: 
        
        # 32ms,14.1MB
        a=sorted(strs)
        
        print(a)
        
        if len(strs)==0:
            return ""
        
        s=""
        for i in range(len(a[0])):
            if (a[0][i] == a[-1][i]):
                s+=a[0][i]
                
            else:
                break
                
        return s   

File: leetcode-interview-swift/leetcode-interview-swift/Easy/Strings/longest_common_prefix.swift
//
//  longest_common_prefix.swift
//  leetcode-interview-swift
//
//  Created by Yusuf Turan on 2.03.2021.
//

func longestCommonPrefix(_ strs: [String]) -> String {
	return strs.longestCommonPrefix()
}

extension Collection where Element: StringProtocol {
	func longestCommonPrefix() -> String {
		guard var prefix = first.map({ String($0) }) else { return "" }
		for string in dropFirst() {
			while !string.hasPrefix(prefix) {
				prefix.removeLast()
			}
		}
		return prefix
	}
	//also we can implement lcs for fun
	func longestCommonSuffix() -> String {
		guard var suffix = first.map({ String($0) }) else { return "" }
		for string in dropFirst() {
			while !string.hasSuffix(suffix) {
				suffix.removeFirst()
			}
		}
		return suffix
	}
	
}
